.paw/work/mkdocs-github-pages/ImplementationPlan.md

```diff
@@ -0,0 +1,182 @@
+# MkDocs GitHub Pages Setup Implementation Plan
+
+## Overview
+
+This plan implements automated documentation publishing for the PAW project using MkDocs with Material theme, deploying to GitHub Pages. Documentation source files will live in `/docs` and be automatically built and published to `https://lossyrob.github.io/phased-agent-workflow` whenever changes are merged to main.
+
+## Current State Analysis
+
+- **No `/docs` folder exists** - Documentation currently lives in root-level Markdown files (README.md, DEVELOPING.md, specifications)
+- **No GitHub Pages configured** - API returns 404; no `gh-pages` branch exists
+- **Two existing workflows** in `.github/workflows/`:
+  - `pr-checks.yml` - Runs tests on PRs (triggers on `src/`, `agents/`, `scripts/`)
+  - `release.yml` - Handles version tagging and VSIX packaging
+- **`.gitignore` includes `.cache/`** but not `site/` (MkDocs build output)
+
+### Key Discoveries:
+- Existing workflows use `@v4` action versions, `ubuntu-latest` runners, and descriptive comments (`.github/workflows/pr-checks.yml:1-4`)
+- Configuration files follow root-level placement pattern (`package.json`, `tsconfig.json`)
+- Official Material for MkDocs deployment pattern uses `mkdocs gh-deploy --force` which handles `gh-pages` branch creation automatically
+
+## Desired End State
+
+After this plan is complete:
+- Documentation site is live at `https://lossyrob.github.io/phased-agent-workflow`
+- Merges to `main` that include changes to `docs/` or `mkdocs.yml` trigger automatic rebuilds
+- Documentation structure is minimal but functional with search, navigation, and Material theme styling
+- Contributors can add/edit documentation by modifying Markdown files in `/docs`
+
+### Verification:
+1. GitHub Actions workflow completes successfully on merge to main
+2. Site loads at the GitHub Pages URL with correct content
+3. Search returns results for terms in documentation
+4. Navigation displays organized structure
+
+## What We're NOT Doing
+
+- **NOT migrating existing documentation** (README.md, DEVELOPING.md, specifications) into the new structure—that's a future enhancement
```

Comment:
Let's add a phase to actually do this. Don't remove the existing documentation, but generate new docs that hold the information in an organized way on the new doc site. Not for DEVELOPING.md, but for the content in the README, in paw-specification, paw-review-specification, and a dive into the Docs.md of existing paw work items that ensures the docs are up-to-date and have representation in the docs site.